src/test/java/test/listeners/AlterSuiteListenerTest.java

```diff
@@ -78,4 +100,42 @@ public void alter(List<XmlSuite> suites) {
         }
     }
 
+    public static class AlterXmlTestWithParameterInSuiteListener implements IAlterSuiteListener {
+
+        @Override
+        public void alter(List<XmlSuite> suites) {
+            XmlSuite suite = suites.get(0);
+            List<XmlTest> tests = suite.getTests();
+            XmlTest test = tests.get(0);
+
+            List<XmlTest> newXmlTests = new ArrayList<>();
+            XmlTest newXmlTest = (XmlTest) test.clone();
+            newXmlTest.setName("name_1");
+            newXmlTest.addParameter("param", "1");
+            newXmlTests.add(newXmlTest);
+
+            newXmlTest = (XmlTest) test.clone();
+            newXmlTest.setName("name_2");
+            newXmlTest.addParameter("param", "2");
+            newXmlTests.add(newXmlTest);
+
+            suite.setTests(newXmlTests);
+        }
+    }
+
+    public static class AlteredXmlSuiteReadListener implements ISuiteListener {
+
+        @Override
+        public void onStart(ISuite suite) {
+            XmlSuite xmlSuite = suite.getXmlSuite();
+            List<XmlTest> tests = xmlSuite.getTests();
+            int i = 1;
+            for (XmlTest xmlTest:
+                 tests) {
+                Assert.assertEquals(xmlTest.getParameter("param"), String.valueOf(i));
```

Comment:
@aditya-qapitol - You should basically collect all the listeners that you need and expose them to your test case (the one that extends `SimpleBaseTest`) wherein you can add assertion rather than here.

Reply:
Please check updated PR